### ISSUE TITLE
Having a problem using info file with data addr[-addr] and dec addr[-addr]

### ISSUE THREAD

**Issue by phillipeaton** [Arakula/f9dasm#14]
Hello,

I have a binary that was compiled from partial source that I have and I'm trying to recreate it using f9dasm.

The original code looks like this:
```
        FCB     +09,+11,-11,-09
        FCB     +10,-10,+01,-01
        FCB     -21,-12,+08,+19
        FCB     +21,+12,-08,-19
        FCB     +10,+10,+11,+09
        FCB     -10,-10,-11,-09
```
If I use an info file with no formatting, I get: 
```
M0000   ROL     M000B                    *0000: 09 0B          '..'
M0002   BITB    MF70A                    *0002: F5 F7 0A       '...'
M0005   LDB     M01FF                    *0005: F6 01 FF       '...'
M0008   ADDB    [,S]                     *0008: EB F4          '..'
M000A   ASL     M0013                    *000A: 08 13          '..'
M000C   FCB     $15                      *000C: 15             '.'
M000D   INC     M00F8                    *000D: 0C F8          '..'
M000F   STD     $0A,X                    *000F: ED 0A          '..'
        DEC     M000B                    *0011: 0A 0B          '..'
M0013   ROL     M00F6                    *0013: 09 F6          '..'
        LDB     MF5F7                    *0015: F6 F5 F7       '...'
```
If I use `data 0000-0018`, I get:
```
M0000   FCB     $09                      *0000: 09             '.'
M0001   FCB     $0B                      *0001: 0B             '.'
M0002   FCB     $F5                      *0002: F5             '.'
M0003   FCB     $F7                      *0003: F7             '.'
M0004   FCB     $0A                      *0004: 0A             '.'
M0005   FCB     $F6                      *0005: F6             '.'
M0006   FCB     $01,$FF                  *0006: 01 FF          '..'
M0008   FCB     $EB,$F4                  *0008: EB F4          '..'
M000A   FCB     $08,$13                  *000A: 08 13          '..'
M000C   FCB     $15                      *000C: 15             '.'
M000D   FCB     $0C,$F8                  *000D: 0C F8          '..'
M000F   FCB     $ED                      *000F: ED             '.'
M0010   FCB     $0A,$0A,$0B,$09,$F6,$F6  *0010: 0A 0A 0B 09 F6 F6 '......'
        FCB     $F5,$F7                  *0016: F5 F7          '..'
```
If I use `data 0000-0018 and decimal 0000-0018` , I get the same as no formatting again:
```
M0000   ROL     M000B                    *0000: 09 0B          '..'
M0002   BITB    MF70A                    *0002: F5 F7 0A       '...'
M0005   LDB     M01FF                    *0005: F6 01 FF       '...'
M0008   ADDB    [,S]                     *0008: EB F4          '..'
M000A   ASL     M0013                    *000A: 08 13          '..'
M000C   FCB     21                       *000C: 15             '.'
M000D   INC     M00F8                    *000D: 0C F8          '..'
M000F   STD     10,X                     *000F: ED 0A          '..'
        DEC     M000B                    *0011: 0A 0B          '..'
M0013   ROL     M00F6                    *0013: 09 F6          '..'
        LDB     MF5F7                    *0015: F6 F5 F7       '...'
```
If I use `dec 0000-0018` without `data`, I get regular code, but with decimal format numbers, as expected.

Whilst I don't expect to be able to get the 4 by 6 table of numbers, as per the original source, from what I read in the Info File spec, I should able to get the output as data as decimal.

Any clues that could help?

Thanks for reading.

**Comment by Arakula** [comment]
`data 0-18` followed by `+dec 0-18` (note the +) should do the trick.

**Comment by Arakula** [comment]
To get the original 4 by 6 layout, you'd need to get rid of the labels (which, of course, might "magically disappear", Trump-style, if you declare other areas as data areas; if not, adding something like `label 0001 M0000+1` might help), and then add
```
break 0004
break 0008
break 000c
break 0010
break 0014
````
... but denoting the positive integers with a '+' prefix isn't implemented, sorry.

**Comment by phillipeaton** [comment]
Not perfect showing as an unsigned number, but it's a special case in the code, so I can manually comment round it. 

The 'break' capability is a nice bonus.

```
        FCB     9,11,245,247             *0180: 09 0B F5 F7    '....'
        FCB     10,246,1,255             *0184: 0A F6 01 FF    '....'
        FCB     235,244,8,19             *0188: EB F4 08 13    '....'
        FCB     21,12,248,237            *018C: 15 0C F8 ED    '....'
        FCB     10,10,11,9               *0190: 0A 0A 0B 09    '....'
        FCB     246,246,245,247          *0194: F6 F6 F5 F7    '....'
```

Just for info, what I'm doing is extracting a game binary from a FLEX CMD file, creating a disassembly of it, ready to remap and assemble onto the Vectrex memory map, to which I'll then be adding video and sound routines written in Forth. 

I have to start with a good disassembly, though, so many thanks for your assistance!

**Comment by phillipeaton** [comment]
f9dasm can't do this, but dasmfw can! Perfect would be a leading zero and leading sign indicator for +ve numbers also, but it's close enough 🙂
```
DIRECT  FCB     9,11,-11,-9             ; +09,+11,-11,-09
        FCB     10,-10,1,-1             ; +10,-10,+01,-01
        FCB     -21,-12,8,19            ; -21,-12,+08,+19
        FCB     21,12,-8,-19            ; +21,+12,-08,-19
        FCB     10,10,11,9              ; +10,+10,+11,+09
        FCB     -10,-10,-11,-9          ; -10,-10,-11,-09
```
Like!

**Comment by Arakula** [comment]
I've uploaded a new dasmfw version, BTW. Including improved documentation, which should be complete now, and (hope)fully working **option cref** .
A .zip file including the executable for Windows is now available on https://www.hermannseib.com/programs/beta/ - better than uploading it to github.

**Comment by phillipeaton** [comment]
Thanks for the updates. I've been running through what's listed in the new help file in the binary you posted a couple of days ago. The CREF I tried, but it didn't seem to work, now I know why. The deflabel works great. I tried using cross-referencing and the result I got wasn't quite an expected, I need to look at it some more and understand it better. dasmfw seems good enough for me to move over to from f9dasm, so I'm doing a file/directory restructure at the same time. 🙂
